crd-generator/api/src/main/java/io/fabric8/crd/generator/CRDGenerationInfo.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.crd.generator;
+
+import io.fabric8.crd.generator.visitor.ClassDependenciesVisitor;
+
+import java.io.File;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CRDGenerationInfo {
+
+  static final CRDGenerationInfo EMPTY = new CRDGenerationInfo();
+  private final Map<String, Map<String, CRDInfo>> crdNameToVersionToCRDInfoMap = new HashMap<>();
+
+  public Map<String, CRDInfo> getCRDInfos(String crdName) {
+    return crdNameToVersionToCRDInfoMap.get(crdName);
+  }
+
+  public Map<String, Map<String, CRDInfo>> getCRDDetailsPerNameAndVersion() {
+    return crdNameToVersionToCRDInfoMap;
+  }
+
+  void add(String crdName, String version, URI fileURI) {
+    crdNameToVersionToCRDInfoMap.computeIfAbsent(crdName, k -> new HashMap<>())
+      .put(version, new CRDInfo(crdName, version, new File(fileURI).getAbsolutePath(), ClassDependenciesVisitor.getDependentClassesFromCRDName(crdName)));
+  }
+
+  public int numberOfGeneratedCRDs() {
+    final int[] result = new int[1];
+    crdNameToVersionToCRDInfoMap.forEach((k, v) -> result[0] += v.size());
+    return result[0];
```

Comment:
Umm, I haven't checked the details but why do we need the `result` to be an array here? Do you think this can be simplified like this:
```java
crdNameToVersionToCRDInfoMap.values().stream()
      .map(Map::size)
      .reduce(0, Integer::sum);
```

Reply:
Good point!